### PR TITLE
execute callback with error if signing is not possible

### LINKED ISSUE #211: getSignedUrl requires `private_key`
To generate a signed url, we need a private_key: https://developers.google.com/storage/docs/accesscontrol#Signed-URLs & https://github.com/GoogleCloudPlatform/gcloud-node/blob/v0.6.0/lib/storage/index.js#L267

We get this if a user provides a credentials object or path to a keyfile, but not in GCE or in the future, GAE. Is it going to be possible to get the private key in those environments?

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #211

If the required properties to create a signed URL or policy aren't available, the user's callback will be executed with an error indicating so.

### CODE REVIEW

File: lib/storage/file.js [hunk 1]
```diff
@@ -1061,7 +1061,19 @@ File.prototype.getSignedPolicy = function(options, callback) {
 
   makeAuthorizedRequest_.getCredentials(function(err, credentials) {
     if (err) {
-      callback(err);
+      var signingError = new Error('Signing failed. See `error` property.');
+      signingError.error = err;
+      callback(signingError);
```
Comment (callmehiphop) [inline lib/storage/file.js]: Since this is happening in more than one place, do you think it would be worth consolidating the logic into something re-usable? like.. `callback(new SigningError(err));`?

Comment (stephenplusplus) [inline lib/storage/file.js]: I think this is a case where that would be more engineering than its worth. Sub-classing an Error is quite a noisy process. However, it would be nice to go over our errors sometime with something like https://github.com/floatdrop/create-error-class and maybe lump them into a "lib/common/errors.js" file or even one for each module: "lib/storage/errors.js". Again though, I'm not even sure *that's* worth the effort.

Comment (callmehiphop) [inline lib/storage/file.js]: IMO if it makes our code base even a little bit cleaner I think it's worth it. I'll open a separate issue

Comment (stephenplusplus) [inline lib/storage/file.js]: SG.

Comment (stephenplusplus) [inline lib/storage/file.js]: We could just undo this change and pass the original error? It's only these situations where many API calls are made that the user doesn't realize. When we get a fail along the way, it would likely make no sense to the user.

Comment (callmehiphop) [inline lib/storage/file.js]: No, I think it's good to be more informative about where it failed.
